pkg/local_object_storage/metabase/lock.go

```diff
@@ -201,17 +202,20 @@ func associatedWithTypedObject(currEpoch uint64, metaCursor *bbolt.Cursor, idObj
 				}
 			}
 
-			return true
+			var associateID oid.ID
+			copy(associateID[:], mainObj)
+			return true, associateID
 		}
 	}
 
-	return false
+	return false, oid.ID{}
 }
 
 // checks if specified object is locked in the specified container.
 func objectLocked(tx *bbolt.Tx, currEpoch uint64, metaCursor *bbolt.Cursor, idCnr cid.ID, idObj oid.ID) bool {
-	if associatedWithTypedObject(currEpoch, metaCursor, idObj, object.TypeLock) {
-		return true
+	locked, lockID := associatedWithTypedObject(currEpoch, metaCursor, idObj, object.TypeLock)
```

Comment:
> Lock can be deleted manually

why? https://github.com/nspcc-dev/neofs-api/blob/fc8584b766d6b544337fc055dca34c29b0155d31/object/service.proto#L92-L93

Reply:
Control service.

Reply:
In fact we have a test that does exactly that, that's how it was discovered initially.